tuf_conformance/test_file_download.py

```diff
@@ -26,244 +13,70 @@ def test_client_downloads_expected_file(
-    # and the file is ready for download. The client updates.
-    client.refresh(init_data)
-
-    # Check that the remote file is the one we expect before
-    # the client downloads.
-    assert os.path.isfile(target_file_path)
-    with open(target_file_path) as f:
-        assert f.read() == file_contents_str
-
-    # Check that the client has not downloaded any files yet
-    assert client.get_last_downloaded_target() == ""
-
-    # Client downloads the file by referencing the name specified
-    # in the targets metadata.
-    url_prefix = get_url_prefix(server_process_handler, client)
-    client.download_target(init_data, target_base_name, target_base_url=url_prefix)
-    # Sanity check that the client has downloaded the file
-    assert client.get_last_downloaded_target() == os.path.join(
-        client._target_dir.name, target_base_name
-    )
-
-    # Check that the file the client download is the expected.
-    with open(client.get_last_downloaded_target(), "r") as last_download_file:
-        assert last_download_file.read() == file_contents_str
-
-    # Now we replace the remote file with an incorrect one.
-    # Here we test that even though the repos targets
-    # metadata is bumped, the client still checks the hashes
-    # and length of files before downloading files.
-
-    # Create a file in the repo with the same name but different contents
-    # The client should not download this.
-    malicious_file_contents_str = "malicious data - should not download"
-    new_target_file = open(target_file_path, "w")
-    new_target_file.write(malicious_file_contents_str)
-    new_target_file.close()
-
-    # Update target version target to repository without
-    # updating the file hashes and length
-    repo.targets.version += 1
-    repo.update_snapshot()
+    # Create legitimate test artifacts
+    target_path_1 = "target_file.txt"
+    target_content_1 = b"target file contents"
+    target_path_2 = "another_target_file.txt"
+    target_content_2 = b"content"
+    repo.add_target(Targets.type, target_content_1, target_path_1)
+    repo.add_target(Targets.type, target_content_2, target_path_2)
 
     # Client updates
-    client.refresh(init_data)
+    assert client.refresh(init_data) == 0
 
-    # Client downloads the same file as it did earlier.
-    client.download_target(init_data, target_base_name, target_base_url=url_prefix)
+    # Download one of the artifacts
+    assert client.download_target(init_data, target_path_1) == 0
+    with open(client.get_last_downloaded_target(), "rb") as last_download_file:
+        assert last_download_file.read() == target_content_1
 
-    # Now we check that the client has not downloaded the
-    # malicious file which is expected. We check the
-    with open(client.get_last_downloaded_target(), "r") as last_download_file:
-        assert last_download_file.read() == file_contents_str
-
-
-def test_repository_substitutes_target_file_and_bumps_10_times(
```

Comment:
ah, this test was actually removed: I don't think it was adding any real coverage.